Problem: "deposit_amount" test failed occationally

```
_____________________________ test_deposit_amount ______________________________

addresses = <pytests.conftest.TestAddresses object at 0x7fd715660400>

    @pytest.mark.withfee
    def test_deposit_amount(addresses):
        'execute a deposit amount to self, check staking state change'
        bonded1 = int(rpc.staking.state(addresses.unbonded_staking)['bonded'])
        txid = rpc.staking.deposit_amount(addresses.unbonded_staking, 100000000)
        wait_for_tx(rpc, txid)
        rpc.wallet.sync()
        bonded2 = int(rpc.staking.state(addresses.unbonded_staking)['bonded'])
>       assert bonded1 + 100000000 == bonded2
E       assert (0 + 100000000) == 0

pytests/test_small.py:16: AssertionError
```

--- a/client-rpc/src/rpc/staking_rpc.rs
+++ b/client-rpc/src/rpc/staking_rpc.rs
@@ -245,7 +245,7 @@ where
                 tx_pending,
             )
             .map_err(to_rpc_error)?;
-        Ok(hex::encode(tx_id))
+        Ok(hex::encode(transaction.tx_id()))
     }
 
     fn state(&self, address: StakedStateAddress) -> Result<StakedState> {

